iceoryx2-pal/posix/src/freebsd/mman.rs

```diff
@@ -33,12 +38,95 @@ pub unsafe fn munlockall() -> int {
     crate::internal::munlockall()
 }
 
+unsafe fn remove_leading_path_separator(value: *const c_char) -> *const c_char {
+    if *value as u8 == PATH_SEPARATOR {
+        value.add(1)
+    } else {
+        value
+    }
+}
+
+unsafe fn shm_file_path(name: *const c_char, suffix: &[u8]) -> [u8; MAX_PATH_LENGTH] {
+    let name = remove_leading_path_separator(name);
+
+    let mut state_file_path = [0u8; MAX_PATH_LENGTH];
+
+    // path
+    state_file_path[..SHM_STATE_DIRECTORY.len()].copy_from_slice(SHM_STATE_DIRECTORY);
+
+    // name
+    let mut name_len = 0;
+    for i in 0..usize::MAX {
+        let c = *(name.add(i) as *const u8);
+
+        state_file_path[i + SHM_STATE_DIRECTORY.len()] = if c == b'/' { b'\\' } else { c };
+        if *(name.add(i)) == 0i8 {
+            name_len = i;
+            break;
+        }
+    }
+
+    // suffix
+    for i in 0..suffix.len() {
+        state_file_path[i + SHM_STATE_DIRECTORY.len() + name_len] = suffix[i];
+    }
```

Comment:
```suggestion
    let start_index = SHM_STATE_DIRECTORY.len() + name_len;
    state_file_path[start_index..start_index+suffix.len()].copy_from_slice(&suffix);
```
No guarantees it will compile :)